app/uk/gov/hmrc/exports/controllers/testonly/TestOnlyController.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.controllers.testonly
+
+import play.api.libs.json.Json
+import play.api.mvc.{Action, ControllerComponents, Request}
+import uk.gov.hmrc.exports.controllers.RESTController
+import uk.gov.hmrc.exports.models.Eori
+import uk.gov.hmrc.exports.models.declaration._
+import uk.gov.hmrc.exports.models.declaration.notifications.{NotificationDetails, ParsedNotification}
+import uk.gov.hmrc.exports.models.declaration.submissions.{Submission, SubmissionRequest, Action => SubmissionAction}
+import uk.gov.hmrc.exports.models.declaration.submissions.SubmissionStatus.{ACCEPTED, ADDITIONAL_DOCUMENTS_REQUIRED, SubmissionStatus}
+import uk.gov.hmrc.exports.repositories.{DeclarationRepository, ParsedNotificationRepository, SubmissionRepository}
+import uk.gov.hmrc.exports.util.ExportsDeclarationBuilder
+
+import java.time.{LocalDateTime, ZoneId, ZonedDateTime}
+import java.util.UUID
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Random
+
+@Singleton
+class TestOnlyController @Inject()(
+  declarationRepository: DeclarationRepository,
+  submissionRepository: SubmissionRepository,
+  parsedNotificationRepository: ParsedNotificationRepository,
+  cc: ControllerComponents
+)(implicit executionContext: ExecutionContext)
+    extends RESTController(cc) {
+  import TestOnlyController._
+
+  implicit val format = Json.format[CreateSubmitDecDocumentsRequest]
+
+  def createSubmittedDecDocuments(): Action[CreateSubmitDecDocumentsRequest] = Action.async(parsingJson[CreateSubmitDecDocumentsRequest]) {
+    implicit request =>
+      for {
+        declaration <- declarationRepository.create(createDeclaration())
+        notification <- saveNotification(createNotification(declaration))
+        maybeDmsDocNotification <- optionallySaveDmsDocNotification(declaration, notification)
+        _ <- submissionRepository.save(createSubmission(declaration, notification))
+      } yield {
+        val status = if (notification != maybeDmsDocNotification) "CREATED WITH DMSDOC" else "CREATED"
+        val conRef = declaration.consignmentReferences
+        Ok(s"EORI:${request.body.eori}, LRN:${conRef.map(_.lrn).getOrElse("")}, MRN:${conRef.flatMap(_.mrn).getOrElse("")}, $status")
```

Comment:
Not sure it is relevant or not as it's just a test endpoint, but could not this be a Json response? Something like...
```
Json.obj("eori" -> request.body.eori, ...)
```

Reply:
There is no requirement for this endpoint to return anything, I've only included this response for debug purposes really.